jdaviz/core/template_mixin.py

```diff
@@ -3525,17 +3532,34 @@ def layer_in_flux_viewer(data):
             return data.label in [lyr.layer.label for lyr in self.flux_viewer.layers]  # noqa E741
 
         def is_trace(data):
-            return hasattr(data, 'meta') and 'Trace' in data.meta
+            return 'Trace' in getattr(data, 'meta', [])
 
         def not_trace(data):
             return not is_trace(data)
 
         def is_image(data):
             return len(data.shape) == 2
 
+        def is_image_not_spectrum(data):
+            return (is_image(data)
+                    and not getattr(data.coords, 'is_spectral', True))
+
         def is_cube(data):
             return len(data.shape) == 3
 
+        def is_cube_or_image(data):
+            return len(data.shape) >= 2
+
+        def is_spectrum(data):
+            return (len(data.shape) == 1
+                    and data.coords is not None
+                    and getattr(data.coords, 'is_spectral', True))
+
+        def is_2d_spectrum_or_trace(data):
+            return (data.ndim == 2
+                    and data.coords is not None
+                    and getattr(data.coords, 'is_spectral', True)) or 'Trace' in data.meta
```

Comment:
I'm not sure if this is the best way to determine if a data collection is a spectrum or not (as opposed to an image in the 2d case, or a trace or light curve in the 1d case).  @pllim suggested checking the presence and dimensions of the flux component, which should work for the trace case, but might not if we ever want to support light curves side-by-side.  For now we don't need to worry about light curves though, so if there is a case where the `data.coords` check would fail (uncalibrated spectra, for example), that might be the better way to go for now?

Reply:
My other suggestion was to inject such data type into Data metadata on load (in the parser). Then the lookup would always be O(1).

Reply:
that is probably the cheapest and simplest since we likely have access to the actual specutils object at that piont, the only downside is that we then _have_ to remember to set that metadata for any future tests, etc.  I'm honestly not sure which is the best - I'll see if anyone else has thoughts and then we can just pick one and cross our fingers it was the right choice.

Reply:
Setting the data type in the metadata sounds good to me. It would make things more explicit than checking coords and shape, although maybe we would keep that as a fallback in case the data type is not set.

Reply:
FWIW in three years when specutils 2 is out we'll have `spectral_axis_index` in the `meta` for anything loaded as a `Spectrum1D` - I'm ok with adding something else to the `meta` for now if it makes this check easier.

Reply:
I'm worried that adding something to metadata might blow up a bit in scope - any objections to leaving as is now (unless someone can think of a case that fails this test) and revisit some sort of metadata addition as we generalize parsers, etc, down the road?

Reply:
That's fine with me, I was just agreeing that could be an ok strategy, not demanding we implement that strategy.